pyseir/inference/infer_rt.py

```diff
@@ -626,10 +712,266 @@ def infer_all(self, plot=True, shift_deaths=0):
+            # dates.append(predicted_days)
+        logging.info("forecasts")
+        logging.info(forecasts)
+        logging.info("dates")
+        logging.info(dates)
+        # check if dictionary of scalers works
+
+        return
+
+    @staticmethod
+    def get_reshaped_X(input_X, n_batch, X_scaler):
+        i = input_X.reshape(n_batch, input_X.shape[0], input_X.shape[1])
+        output_df = pd.DataFrame(np.squeeze(input_X))
+        # original_df = pd.DataFrame(X_scaler.inverse_transform(output_df))
+        return original_df
+
+    @staticmethod
+    def build_model(
+        MASK_VALUE, epochs, n_batch, hidden_layer_dimensions, dropout, final_train_X, final_train_Y
+    ):
+        patience = 50
+        validation_split = 0.1
+        model = Sequential()
+        model.add(
+            Masking(
+                mask_value=MASK_VALUE,
+                batch_input_shape=(n_batch, final_train_X.shape[1], final_train_X.shape[2]),
+            )
+        )
+        model.add(
+            LSTM(
+                hidden_layer_dimensions,
+                batch_input_shape=(n_batch, final_train_X.shape[1], final_train_X.shape[2]),
+                stateful=True,
+                return_sequences=True,
+            )
+        )
+        model.add(
+            LSTM(
+                hidden_layer_dimensions,
+                batch_input_shape=(n_batch, final_train_X.shape[1], final_train_X.shape[2]),
+                stateful=True,
+            )
+        )
+        model.add(Dropout(dropout))
+        model.add(Dense(final_train_Y.shape[1]))
+        es = EarlyStopping(monitor="val_loss", mode="min", verbose=1, patience=patience)
+        model.compile(loss="mean_squared_error", optimizer="adam")
+        history = model.fit(
+            final_train_X,
+            final_train_Y,
+            epochs=epochs,
+            batch_size=n_batch,
+            verbose=1,
+            shuffle=False,
+            validation_split=validation_split,
+            callbacks=[es],
+        )
+        logging.info("fit")
+        logging.info(history.history["loss"])
+        logging.info(history.history["val_loss"])
+        plot = True
+        if plot:
+            plt.close("all")
+            logging.info("plotting")
+            plt.plot(history.history["loss"], color="blue", linestyle="solid", label="Train Set")
+            logging.info("plotted history")
+            plt.plot(
+                history.history["val_loss"],
+                color="green",
+                linestyle="solid",
+                label="Validation Set",
+            )
+            logging.info("plotted more")
+            plt.legend()
+            plt.xlabel("Epochs")
+            plt.ylabel("RMSE")
+            plt.savefig("lstm_loss_final.png")
+            plt.close("all")
+
+        return model, history
+
+    @staticmethod
+    def get_X_Y(sample_list, PREDICT_DAYS, PREDICT_VARIABLE, MASK_VALUE):
```

Comment:
I'd prefer these variables to be all lower case for consistency. What do you think about making a configuration data class that you could pass around?

```
import dataclasses

@dataclasses.dataclass
class LSTMModelConfig:
    batch_size: int = 1
    predict_days: int = ...


model_config = LSTMModelConfig()
```

I think all of the variables you have capitalized could go in the config.  The nice thing is that you can specify defaults, but then easily override values:
```
model_config = LSTMModelConfig(predict_days=100)
```